### ISSUE TITLE
Questions about the regression of delta

### ISSUE THREAD

**Issue by kosuke55** [OYMiss/faster-rcnn#1]
Hi, thanks for useful work. I'm not sure about the calculation of the delta regression.
In the [rcnn_loss_layer function](https://github.com/OYMiss/faster-rcnn/blob/f219ebb92ee2964a6b8348e64ee21d42f0cc1403/layers/rcnn_loss_layer.py#L12-L43), I think the shape of delta is (p, c, 4) and p is the sum of all proposals in all batches.
For the i-th batch, find the index of a particular class, then accesses delta using the index
https://github.com/OYMiss/faster-rcnn/blob/f219ebb92ee2964a6b8348e64ee21d42f0cc1403/layers/rcnn_loss_layer.py#L12-L43
```
x = fg_i[index_j]
pred_delta_j = delta[x, j]
```
But I think the index of x here is to indicate the corresponding proposals in the batch of i.
In other words, since delta has a length of P that concatenates all the batches, it does not access the batch number i using x. I think I need to stagger the idx to account for batches to access the correct delta, but what about that?

**Comment by OYMiss** [comment]
Thank you and sorry for my mistake. 

I think the [new commit](https://github.com/OYMiss/faster-rcnn/commit/0cb46f721179456da22055cadfc552d7e377d422) will fix this.

**Comment by kosuke55** [comment]
@OYMiss 
Thank you for taking the time to respond. I think it's okay too.
